docs: Add info on GitHub filter push down (#438)
spiceai/docs: spiceaidocs/docs/components/data-connectors/github.md

- `owner` - Required. Specifies the owner of the GitHub repository.
- `repo` - Required. Specifies the name of the GitHub repository.

### Filter Push Down

GitHub queries support a `github_query_mode` parameter, which can be set to either `auto` or `search` for the following types:

> - **Issues**: Defaults to `auto`. Query filters are only pushed down to the GitHub API in `search` mode.

Reviewer: nit: maybe this should be a table.

> Querying GitHub supports a `github_query_mode` parameter value of either `auto` or `search` in the following connectors:
> 
> - Issues; defaults to `auto`, query filters are only pushed down in `search` mode.
> - Pull Requests; defaults to `auto`, query filters are only pushed down in `search` mode.

Reviewer 2: ```suggestion
GitHub queries support a `github_query_mode` parameter, which can be set to either `auto` or `search` for the following types:

- **Issues**: Defaults to `auto`. Query filters are only pushed down to the GitHub API in `search` mode.
- **Pull Requests**: Defaults to `auto`. Query filters are only pushed down to the GitHub API in `search` mode.
```

> The Commits connector supports only `auto` mode, with query filter push down enabled for the `committed_date` column. `commited_date` supports exact matches, or greater/less than matches for dates provided in [ISO8601](https://www.iso.org/iso-8601-date-and-time-format.html) format, like `WHERE committed_date > '2024-09-24'`.

Reviewer 2: ```suggestion
Commits only supports `auto` mode. Query with filter push down is only enabled for the `committed_date` column. `commited_date` supports exact matches, or greater/less than matches for dates provided in [ISO8601](https://www.iso.org/iso-8601-date-and-time-format.html) format, like `WHERE committed_date > '2024-09-24'`.
```

When set to `search`, Issues and Pull Requests will use the GitHub [Search API](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests) for improved filter performance when querying against the columns:

- `author` and `state`; supports exact matches, or NOT matches. For exmaple, `WHERE author = 'peasee'` or `WHERE author <> 'peasee'`.
- `body` and `title`; supports exact matches, or LIKE matches. For example, `WHERE body LIKE '%duckdb%'`.
- `updated_at`, `created_at`, `merged_at` and `closed_at`; supports exact matches, or greater/less than matches with dates provided in [ISO8601](https://www.iso.org/iso-8601-date-and-time-format.html) format. For example, `WHERE created_at > '2024-09-24'`.

All other filters are supported when `github_query_mode` is set to `search`, but cannot be pushed down to the GitHub API for improved performance.

:::warning[Limitations]

- GitHub has a limitation in the Search API where it may return more stale data than the standard API used in the default query mode.

:::

### Querying GitHub Files

:::warning[Limitations]

> - `content` column is included only when acceleration is enabled.

Reviewer 2: ```suggestion
- `content` column is fetched only when acceleration is enabled.
```
